src/main/java/org/jnode/fs/xfs/XfsObject.java

```diff
@@ -97,4 +124,69 @@ public long getUInt32(int relativeOffset) {
     public long getInt64(int relativeOffset) {
         return BigEndian.getInt64(data, offset + relativeOffset);
     }
+
+    /**
+     * Gets the UUID value.
+     *
+     * @param offset the offset to read from.
+     * @return the uuid value.
+     */
+    protected String readUuid(int offset) {
+                 return Long.toHexString(getUInt32(offset))
+                + "-" + Long.toHexString(getUInt16(offset + 4))
+                + "-" + Long.toHexString(getUInt16(offset + 6))
+                + "-" + Long.toHexString(getUInt16(offset + 8))
+                + "-" + Long.toHexString(getUInt16(offset + 10));
+    }
+
+    /**
+     * Converts ascii to hex.
+     *
+     * @param asciiString value to convert to hex.
+     * @return the hex value.
+     */
+    public static long asciiToHex(String asciiString) {
+        StringBuilder hex = new StringBuilder();
+        for (char c : asciiString.toCharArray()) {
+            hex.append(Integer.toHexString(c).toUpperCase(Locale.ROOT));
+        }
+        return Long.parseLong(hex.toString(), 16);
+    }
+
+    /**
+     * Converts hex to ascii.
+     *
+     * @param hexString value to convert to ascii.
+     * @return the ascii value.
+     */
+    public static String hexToAscii(String hexString) {
+        try {
+            StringBuilder ascii = new StringBuilder();
+            final Iterable<String> chars = Splitter.fixedLength(2).split(hexString);
+            for (String c : chars) {
+                ascii.append((char) Byte.parseByte(c, 16));
+            }
+            return ascii.toString();
+        } catch (Exception e) {
+            return "Invalid";
+        }
+    }
+
+    /**
+     * Gets magic signature.
+     *
+     * @return the hex value.
+     */
+    public long getMagicSignature()  {
+        return getUInt32(0);
+    }
+
+    /**
+     * Gets signature as ascii.
+     *
+     * @return the ascii value.
+     */
+    public String getAsciiSignature() {
```

Comment:
Instead of getting this function to rely on the default implementation of `getMagicSignature()` which may not always be valid for any implementing subclasses, it would probably be cleaner to pass the signature in as a parameter to this function. It probably doesn't need to be `public` either - it could just be `protected`, it doesn't seem likely that a non-child class will call this function.

Reply:
HI, I followed your suggestion, the function now receives  a long (signature) and will return a string ascii.